android/src/main/java/com/swmansion/rnscreens/ScreenContainer.java

```diff
@@ -352,14 +356,16 @@ protected void performUpdate() {
       }
     }
 
-    // detect if we are "transitioning" based on the number of active screens
-    int activeScreens = 0;
+    boolean transitioning = true;
+
     for (int i = 0, size = mScreenFragments.size(); i < size; i++) {
-      if (isScreenActive(mScreenFragments.get(i))) {
-        activeScreens += 1;
+      ScreenFragment screenFragment = mScreenFragments.get(i);
+      Screen.ActivityState active = getActivityState(screenFragment);
+      if (active == Screen.ActivityState.ON_TOP) {
```

Comment:
Why is the activity state variable called "active" – considering the change made here I'd prefer to avoid further confusion and use one term over the other. Also I think we don't need this variable here at all

```suggestion
      if (getActivityState(screenFragment) == Screen.ActivityState.ON_TOP) {
```